system/lib/pthread/pthread_create.c

```diff
@@ -196,7 +186,12 @@ void _emscripten_thread_exit(void* result) {
   self->result = result;
 
   // Run any handlers registered with pthread_cleanup_push
-  __run_cleanup_handlers();
+  while (self->cancelbuf) {
+    void (*f)(void *) = self->cancelbuf->__f;
+    void *x = self->cancelbuf->__x;
+    self->cancelbuf = self->cancelbuf->__next;
+    f(x);
+  }
```

Comment:
I kind of like the self-documenting nature of having this in a separate function..  but I guess you goal here is to reduce the delta from upstream musl?

Reply:
It was indeed to reduce the delta from musl (+ I wasn't sure if it was already in-lined by the compiler, since it called `__pthread_self()` again in that function).

Reply:
OK seems reasonable to try to match the musl code I guess.